app/javascript/packs/advanced_search.js

```diff
@@ -0,0 +1,54 @@
+var checkBoxes = document.querySelectorAll(".form-check-input");
+
+var fabrics = document.querySelectorAll("#fabric");
+
+const advancedSearch = () => {
+  for (const check of checkBoxes) {
+    check.addEventListener("change", function () {
+      let attributes = getCheckedBoxesValues();
+
+      if (attributes.length == 0) {
+        ShowAllCards();
+        return;
+      }
+      fabrics.forEach((fabric) => {
+        if (checkIfFabricContainsAttr(fabric.textContent, attributes)) {
+          ShowCard(fabric);
+        } else {
+          hideCard(fabric);
+        }
+      });
+    });
+  }
+};
+
+const checkIfFabricContainsAttr = (fabric, attributes) => {
+  return attributes.some((attr) => fabric.includes(attr));
```

Comment:
@Victor Albanezi no lugar de .some talvez funcione usando a funcao .every
https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/every

MDN Web DocsMDN Web Docs
Array.prototype.every()
O método every() testa se todos os elementos do array passam pelo teste implementado pela função fornecida.
9:25
aquela historia de mostrar os elementos que tem os 2 checkedboxes. e nao apenas 1 quando se seleciona os 2

Gabriel Antonini  9:40 AM
isso no caso de categorias diferentes quando quer filtrar os elementos tendo os 2 attributos
9:41
vou colocar no PR para ficar mais facil na verdade hehehe
9:41
mas por mim podemos fazer o merge e ir melhorando o comportamento da feature
9:42
evitar conflitos e ja ir vendo css se for preciso da pagina..

Reply:
boa!
alteração feita e commitada já